src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -463,7 +470,29 @@ public void download() {
         }
     }
 
+    public boolean checkSSLHandshake(URLDownload urlDownload) {
+        try {
+            urlDownload.canBeReached();
+        } catch (kong.unirest.UnirestException ex) {
+            if (ex.getCause() instanceof javax.net.ssl.SSLHandshakeException) {
+                if (dialogService.showConfirmationDialogAndWait(Localization.lang("Download file"),
```

Comment:
Do we really need to ask the user, or should we just bypass ssl verification? For example, firefox downloads the linked pdf without any prompt.

Reply:
I think that bypassing SSL verification should let users know. 
We can add an option in the preferences to allow users to select all downloads to bypass the verification. The advantage of this feature that the user does not need to click on the trusted site every time. The disadvantage is that the code changes may be a bit big.

But I found that JabRef uses `URLDownload.bypassSSLVerification()` directly in some places, such as org.jabref.logic.importer.fetcher.IEEE#getURLForQuery, line 253.

Reply:
I think it's okay to show a dialog here. Should not happen that often.